crates/agglayer-storage/src/types/certificate/mod.rs

```diff
@@ -198,12 +200,14 @@ pub enum AggchainDataV1<'a> {
     GenericNoSignature {
         proof: Cow<'a, Proof>,
         aggchain_params: Digest,
+        public_values: Option<Box<AggchainProofPublicValues>>,
     },
 
     GenericWithSignature {
         proof: Cow<'a, Proof>,
         aggchain_params: Digest,
         signature: Cow<'a, Box<Signature>>,
+        public_values: Option<Box<AggchainProofPublicValues>>,
```

Comment:
Unfortunately, adding fields breaks storage binary backwards compatibility. You have to introduce a new enum arm, like `GenericWithPublicValues`. Two arms should not be needed since the new one can have the signature field as an `Option`.

Regarding the arm names, since this enum is only used for versioning storage encoding, I would not mind renaming the arms to just something like `GenericV0`, `GenericV1` and `GenericV2`. I'm afraid that at this rate, we will run out of descriptive names quite soon :laughing:.

Reply:
Filed a PR with tests that would have detected this issue:
* #834